libs/core/algorithms/tests/unit/algorithms/max_element.cpp

```diff
@@ -125,6 +126,7 @@ void max_element_test()
 {
     test_max_element<std::random_access_iterator_tag>();
     test_max_element<std::forward_iterator_tag>();
+    test_max_element_semantics<std::random_access_iterator_tag>();
```

Comment:
Don't you have to forward declare this function? Also, why not calling it for forward-iterators as well?